avm/res/data-factory/factory/main.bicep

```diff
@@ -232,6 +235,23 @@ module dataFactory_integrationRuntimes 'integration-runtime/main.bicep' = [
   }
 ]
 
+module dataFactory_linkedServices 'linked-service/main.bicep' = [
+  for (linkedService, index) in linkedServices: {
+    name: '${uniqueString(deployment().name, location)}-DataFactory-LinkedServices-${index}'
+    params: {
+      dataFactoryName: dataFactory.name
+      name: linkedService.name
+      typeName: linkedService.type
+      typeProperties: contains(linkedService, 'typeProperties') ? linkedService.typeProperties : {}
+      integrationRuntimeName: contains(linkedService, 'integrationRuntimeName')
+        ? linkedService.integrationRuntimeName
+        : 'none'
+      customizedParameter: contains(linkedService, 'parameters') ? linkedService.parameters : {}
+      linkedServiceDescription: linkedService.?linkedServiceDescription ?? 'Linked Service created by avm-res-datafactory-factories'
```

Comment:
I'd suggest to move the defaults to the child module and have Bicep automatically fetch them. You can do this by
1. Move all the defaults for optional parameters you configured here into the parameters in the child (e.g., the `typeProperties` parameter would look like  `param typeProperties object = {}`
2. Change the implementation here to use the `tryGet()` function to automatically pull the defaults if they're not provided (e.g., `parameters: contains(linkedService, 'parameters') ? linkedService.parameters : {}` would be `parameters: linkedService.?parameters` and `description: linkedService.?description ?? 'Linked Service created by avm-res-datafactory-factories'` would be `description: linkedService.?description`

Reply:
It's a neat feature of Bicep that it automatically pulls the default from a referenced template in, if no value was provided (unless the parameter in the referenced template is nullable, of course)